### ISSUE TITLE
MWE for failures in file

### ISSUE THREAD

**Issue by macd** [JuliaDebug/JuliaInterpreter.jl#106]
The first two errors are due to throwing the wrong exception.  The third error has my new favorite error message: ```sigatomic_end called in non-sigatomic region```

```
using JuliaInterpreter
include("utils.jl")

module file_test
    using Test
end

m = file_test

ex = quote
    @test_throws ArgumentError download("good", "ba\0d") # Thrown: UndefVarError
    @test_throws ArgumentError download("ba\0d", "good") # Thrown: UndefVarError
    n = tempname()
    w = open(n, "a")
    write(w, "A")
    flush(w)  #  sigatomic_end called in non-sigatomic region
    #close(w)
    #rm(n)
end

frame = JuliaInterpreter.prepare_thunk(m, ex)
JuliaInterpreter.finish_and_return!(JuliaStackFrame[], frame)
```

**Comment by KristofferC** [comment]
I can't repro the UndefVarErrors:

```
> JuliaInterpreter.finish_and_return!(JuliaStackFrame[], frame)
  Test Passed
      Thrown: ArgumentError
```

**Comment by macd** [comment]
Hmmm, I am using today's master and Ubuntu 18.04 on two different machines. (Intel Skull Canyon NUC and a Lenovo laptop ~4.5 years old) and it fails on both of them.  Any setup issues that I may have bolluxed?

**Comment by KristofferC** [comment]
I'm using Windows and 1.1.0 so maybe it has something to do with that?

**Comment by timholy** [comment]
I tried Ubuntu 18.04 and can't repro the UndefVarErrors. The sigatomic issue will be fixed by #108.
